RxSwift/Observables/Implementations/DefaultIfEmpty.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  DefaultIfEmpty.swift
+//  Rx
+//
+//  Created by sergdort on 23/12/2016.
+//  Copyright © 2016 Krunoslav Zaher. All rights reserved.
+//
+
+import Foundation
+
+class DefaultIfEmptySink<SourceType, O: ObserverType>: Sink<O>, ObserverType where O.E == SourceType {
```

Comment:
We recently had a refactoring change merged.

https://github.com/ReactiveX/RxSwift/pull/1045

I think @tarunon was right, we could make this simpler.